### ISSUE TITLE
Remove ModelTool Specific Package Dependencies

### ISSUE THREAD

**Issue by mehalter** [jpfairbanks/SemanticModels.jl#243]
We should remove the ModelTool Specific Package Dependencies like DiffEq (I think we have done this, at least talked about it) and Petri.jl, and leave it up to the user to add them as they need them. We can create an environment in the `examples` folder to have examples that use these packages.

**Comment by mehalter** [comment]
activate the examples/ env and `add` the local version of the package

```
 $ cd examples
 $ julia --project=.

 julia> # press `]`
 
 (examples) pkg> add ..
then the Manifest.toml should have an entry like
 [[MyPackage]]
 ...
 repo-url = ".."
```

**Comment by jpfairbanks** [comment]
yeah, each subfolder of `examples` should have a Project.toml file that you activate before you run the example, which will install all the necessary dependencies for that example. We might need to split SemanticModels into a SemanticModelsBase which includes all the common types and functions and then packages like

1. SemanticPetri
2. SemanticDiffEq
3. SemanticGraphs
4. SemanticPhysics
5. SemanticBio

that contain the integrations with each domain science's packages
